src/runtime/printer.h

```diff
@@ -32,10 +32,18 @@ class Printer {
     char *buf, *dst, *end;
     void *user_context;
     bool own_mem;
+    char scratch[length <= 256 ? length : 1];
 
     Printer(void *ctx, char *mem = NULL)
         : user_context(ctx), own_mem(mem == NULL) {
-        buf = mem ? mem : (char *)halide_malloc(user_context, length);
+        if (mem != NULL) {
+            buf = mem;
+        } else if (length <= sizeof(scratch)) {
+            buf = scratch;
+        } else {
+            buf = (char *)malloc(length);
```

Comment:
Maybe add a comment explaining why not to use halide_malloc?  Otherwise someone will surely change this in a few weeks :)